_search-plugins/retrieve-specific-fields.md

```diff
@@ -0,0 +1,877 @@
+
+## Searching with `docvalue_fields`
+
+`docvalue_fields` is another parameter you can use in OpenSearch to retrieve specific fields from the index, but it works slightly differently compared to the `fields` parameter. The `docvalue_fields` parameter retrieves details from doc values rather than from the `_source` field, which is more efficient for fields that are not analyzed, like keyword, date, and numeric fields. Doc values are a columnar storage format optimized for efficient sorting and aggregations. It stores the values on disk in a way that is easy to read. When you use `docvalue_fields`, OpenSearch reads the values directly from this optimized storage format. It is useful for retrieving values of fields that are primarily used for sorting, aggregations, and for use in scripts.
+
+To better understand `docvalue_fields` see following example.
+
+
+1. Create index `my_index` with the following mappings:
+
+    ```
+    PUT my_index
+    {
+      "mappings": {
+        "properties": {
+          "title": { "type": "text" },
+          "author": { "type": "keyword" },
+          "publication_date": { "type": "date" },
+          "price": { "type": "double" }
+        }
+      }
+    }
+    ```
+
+2. Index the following documents using the newly created index:
+    ```
+    POST my_index/_doc/1
+    {
+      "title": "OpenSearch Basics",
+      "author": "John Doe",
+      "publication_date": "2021-01-01",
+      "price": 29.99
+    }
+    
+    POST my_index/_doc/2
+    {
+      "title": "Advanced OpenSearch",
+      "author": "Jane Smith",
+      "publication_date": "2022-01-01",
+      "price": 39.99
+    }
+    ```
+3. Retrieve only the `author` and `publication_date` fields using `docvalue_fields`:
+
+    ```
+    POST my_index/_search
+    {
+      "_source": false,
+      "docvalue_fields": ["author", "publication_date"],
+      "query": {
+        "match_all": {}
+      }
+    }
+    ```
+
+Expected response:
+
+```
+{
+  "hits": {
+    "total": {
+      "value": 2,
+      "relation": "eq"
+    },
+    "max_score": 1.0,
+    "hits": [
+      {
+        "_index": "my_index",
+        "_id": "1",
+        "_score": 1.0,
+        "fields": {
+          "author": ["John Doe"],
+          "publication_date": ["2021-01-01T00:00:00.000Z"]
+        }
+      },
+      {
+        "_index": "my_index",
+        "_id": "2",
+        "_score": 1.0,
+        "fields": {
+          "author": ["Jane Smith"],
+          "publication_date": ["2022-01-01T00:00:00.000Z"]
+        }
+      }
+    ]
+  }
+}
+```
+
+### Using `docvalue_fields` with nested objects
+
+In OpenSearch, if you want to retrieve doc values for nested objects, you cannot directly use the `docvalue_fields` parameter because it will return an empty array. Instead, you should use the `inner_hits` parameter with its own `docvalue_fields` property, see following example.
```

Comment:
```suggestion
In OpenSearch, if you want to retrieve doc values for nested objects, you cannot directly use the `docvalue_fields` parameter because it will return an empty array. Instead, you should use the `inner_hits` parameter with its own `docvalue_fields` property, as shown in the following example.
```